app/bundles/LeadBundle/Helper/AnonymizeHelper.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\LeadBundle\Helper;
+
+class AnonymizeHelper
+{
+    public const PRE_DEFINED_DOMAIN       = 'ano.nym';
+    public const PRE_PSEUDONYMIZED_DOMAIN = 'pseudo.nym';
+
+    /**
+     * @param string|bool|null $email
+     */
+    public static function email($email, bool $pseudonymized = false, string $newDomain = self::PRE_DEFINED_DOMAIN): string|false
+    {
+        if (empty($email)) {
+            return '';
+        }
+
+        if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+            return false;
+        }
+        $emailParts = explode('@', $email);
+
+        $toEncrypt= $emailParts[0].time().random_int(0, 999999);
+
+        if ($pseudonymized) {
+            $toEncrypt = $emailParts[0];
+            $newDomain = self::PRE_PSEUDONYMIZED_DOMAIN;
+        }
+        $name       = hash('sha256', $toEncrypt);
+
+        return $name.'@'.$newDomain;
+    }
+
+    /**
+     * @param string|bool|null $text
+     */
+    public static function text($text, bool $pseudonymize = false): string
```

Comment:
Can we rename the method name to something like `anonymizeText` for more verbosity?